### ISSUE TITLE
Refactor keyword-based assessments to accommodate morphology

### ISSUE THREAD

**Issue by nataliashitova** [Yoast/YoastSEO.js#1558]
Current versions of keyword-based assessments rely on exact matches between the keyword and the text. However, now with developed morphology for English exact matches make no sense. Moreover, decisions need to be made on how to deal with multi-word keyphrases (e.g., do we match full or do we allow partial matches).
Hereafter I outline my suggestions on possible adjustments of existing assessments.

**Comment by nataliashitova** [comment]
@jdevalk @omarreiss Could you confirm that is the way to go?

**Comment by omarreiss** [comment]
I Agree with a lot. I see a couple of things @jdevalk needs to decide about.

### 1. Title
> Proposal: Good if an exact match of the keyword is found, OK if all content words from the keyphrase are in the title, BAD otherwise.  

In the proposal it no longer seems to matter if the keyword is at the beginning of the title. Do you agree with this?

### 2. First paragraph
> If the keyphrase has 4 or more content words: GOOD if 3 content words from the keyphrase are matched within one sentence in the introduction, while the rest are found in the neighbour sentences, OK if all content words are present in the first paragraph at all (but not in the same sentence), BAD otherwise.  

I think this is going to be a bit hard to explain. I'd rather not differentiate here and always require all content words to be matched in at least one sentence for a GOOD.

### 3. Subheadings
> Proposal: A subheading is considered to reflect the topic if > half of content words from the keyphrase are used in it. Then, GOOD if 30-75% of subheadings reflect the topic, BAD otherwise.  

I would prefer a definition where a subheading needs to include all of the content words in order to reflect the topic. 

### 4. Urls
> Proposal: If the keyword has 1 or 2 content words: GOOD if all content words are in the URL, OK otherwise. If the keyword has >2 content words: GOOD if > half of content words are in the URL, OK otherwise.

This needs an SEO's perspective.

### 5. Keyphrase length
> Current: GOOD if the keyphrase length is between 1-4, OK if between 5-8, BAD otherwise.
Proposal: Same but count only content words.

This would make it much less strict. From a content words perspective the current calibration is probably already Good: 1-3, OK: 4-6, BAD: otherwise.

**Comment by jdevalk** [comment]
> 1. Title
> > Proposal: Good if an exact match of the keyword is found, OK if all content words from the > keyphrase are in the title, BAD otherwise.
>
> In the proposal it no longer seems to matter if the keyword is at the beginning of the title. Do you > agree with this?

No. Beginning of title matters.

> 2. First paragraph
> > If the keyphrase has 4 or more content words: GOOD if 3 content words from the keyphrase are matched within one sentence in the introduction, while the rest are found in the neighbour sentences, OK if all content words are present in the first paragraph at all (but not in the same sentence), BAD otherwise.

> I think this is going to be a bit hard to explain. I'd rather not differentiate here and always require all content words to be matched in at least one sentence for a GOOD.

Agree with @omarreiss 

> 3. Subheadings
> > Proposal: A subheading is considered to reflect the topic if > half of content words from the keyphrase are used in it. Then, GOOD if 30-75% of subheadings reflect the topic, BAD otherwise.

> I would prefer a definition where a subheading needs to include all of the content words in order to reflect the topic.

That's undoable for larger keyphrases. I think I'm fine with @nataliashitova's suggestion but this needs to be tested on real copy.

> 4. Urls
> > Proposal: If the keyword has 1 or 2 content words: GOOD if all content words are in the URL, OK otherwise. If the keyword has >2 content words: GOOD if > half of content words are in the URL, OK otherwise.

> This needs an SEO's perspective.

I'm fine with this.

> 5. Keyphrase length
> > Current: GOOD if the keyphrase length is between 1-4, OK if between 5-8, BAD otherwise.
Proposal: Same but count only content words.

> This would make it much less strict. From a content words perspective the current calibration is probably already Good: 1-3, OK: 4-6, BAD: otherwise.

I'm fine with less strict for this. So let's go with @nataliashitova's suggestion.
